### PR TITLE
Cat fastq

### PR DESCRIPTION
**Pull request by danilodileo**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/metatdenovo/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/metatdenovo _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

with this new feature it is possible to concatenate the fastq files. 
you need to use the same name in sample column. example:
sample,fastq_1,fastq_2
sample1,example1_R1.fastq.gz,example1_R2fastq.gz
sample1,example2_R1.fastq.gz,example2_R2fastq.gz
sample2,example3_R1.fastq.gz,example3_R2fastq.gz

sample1 and sample2 will be concatenated before all the other steps.

### CODE REVIEW

**Review by erikrikarddaniel — CHANGES_REQUESTED** [review]
You made the PR against the master branch -- 351 commits! Change!

**Comment by danilodileo** [comment]
> You made the PR against the master branch -- 351 commits! Change!

ops!

File: workflows/metatdenovo.nf [hunk 1]
```diff
@@ -156,13 +157,51 @@ workflow METATDENOVO {
...
+    }
+    .groupTuple()
+    .branch {
+        meta, fastq ->
+            single  : fastq.size() == 1
+                return [ meta, fastq.flatten() ]
+            multiple: fastq.size() > 1
+                return [ meta, fastq.flatten() ]
+    }
+    .set { ch_fastq }
```
Comment (erikrikarddaniel) [inline workflows/metatdenovo.nf:174]: Shouldn't this be indented one level more?

File: workflows/metatdenovo.nf [hunk 2]
```diff
@@ -133,6 +133,7 @@ include { BAM_SORT_SAMTOOLS                          } from '../subworkflows/nf-
 include { SUBREAD_FEATURECOUNTS as FEATURECOUNTS_CDS } from '../modules/nf-core/subread/featurecounts/main'
 include { PRODIGAL                                   } from '../modules/nf-core/prodigal/main'
 include { SPADES                                     } from '../modules/nf-core/spades/main'
+include { CAT_FASTQ 				     } from '../modules/nf-core/cat/fastq/main'
```
Comment (erikrikarddaniel) [inline workflows/metatdenovo.nf]: Add space to bring right bracket in line.

File: workflows/metatdenovo.nf [hunk 3]
```diff
@@ -156,13 +157,51 @@ workflow METATDENOVO {
...
 
+    //
+    // MODULE: Concatenate FastQ files from same sample if required
+    //
+    CAT_FASTQ (
+        ch_fastq.multiple
+    )
+    .reads
+    .mix(ch_fastq.single)
+    .set { ch_cat_fastq }
```
Comment (erikrikarddaniel) [inline workflows/metatdenovo.nf:185]: Indent also here?

**Review by erikrikarddaniel — APPROVED** [review]
Some of this is a bit over my head and my suggestions for indentation are perhaps misplaced! ;-)

We should have a test case. Best is probably if we can add a line to the normal sample sheet so it gets tested all the time.

I'm approving, but let's not forget the test case!

File: workflows/metatdenovo.nf (hunk 1 above)
Comment (danilodileo) [reply workflows/metatdenovo.nf:174]: I followed the rnaseq indentation here. I am also not sure

**Comment by danilodileo** [comment]
> Some of this is a bit over my head and my suggestions for indentation are perhaps misplaced! ;-)
> 
> We should have a test case. Best is probably if we can add a line to the normal sample sheet so it gets tested all the time.
> 
> I'm approving, but let's not forget the test case!

I can create an issue

**Pull request merged**